shared/src/ledger/eth_bridge/storage/vote_tracked.rs

```diff
@@ -6,27 +6,21 @@ use crate::types::storage::Key;
 #[allow(missing_docs)]
 pub const PREFIX_KEY_SEGMENT: &str = "eth_msgs";
 
-/// Get the key prefix corresponding to where details of seen [`EthereumEvent`]s
-/// are stored
-pub fn prefix() -> Key {
-    super::prefix()
-        .push(&PREFIX_KEY_SEGMENT.to_owned())
-        .expect("should always be able to construct this key")
-}
-
 const BODY_KEY_SEGMENT: &str = "body";
 const SEEN_KEY_SEGMENT: &str = "seen";
 const SEEN_BY_KEY_SEGMENT: &str = "seen_by";
 const VOTING_POWER_KEY_SEGMENT: &str = "voting_power";
 
-/// Generator for the keys under which details of an [`EthereumEvent`] is stored
-pub struct Keys {
-    /// The prefix under which the details are an [`EthereumEvent`] is stored
+/// Generator for the keys under which details of a vote tracked event is stored
+pub struct Keys<T> {
+    /// The prefix under which the details of a vote tracked event is stored
     pub prefix: Key,
+    _phantom: std::marker::PhantomData<T>,
```

Comment:
```suggestion
    _phantom: std::marker::PhantomData<&'static T>,
```
we don't own values of `T`, no need to trigger the drop checker

Reply:
why do we need to include a type parameter here though? is it being used anywhere? it would be nice if you could use this `T` to derive a new sub-key or smth, but it's not being used that way.

Reply:
Oh nice, I didn't know we could use `'static` here. Re. using a type parameter, it was so that `vote_tracked::Keys` to do with different things (i.e. Ethereum events vs valset updates) would have different types. So could do things like

```rust
impl From<&Hash> for Keys<EthereumEvent>
```

where the returned `keys` struct has a prefix which uses `eth_msgs_prefix()`, a `Keys<ValsetUpd>` or something might still be buildable from a hash but use a `valset_upd_prefix()` for example.

then we can also do stuff like

```rust
let keys: Keys<EthereumEvent> = (&hash).into();
```

Maybe there is a better way to handle this though, can change this.

Reply:
Maybe `T` should be bounded here by `BorshSerialize + BorshDeserialize` as well. I guess there are only specific `T` that should ever be allowed here (right now, just `EthereumEvent`).